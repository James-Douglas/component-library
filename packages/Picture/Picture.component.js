import React from 'react';
import PropTypes from 'prop-types';
import {
  StyledPicture,
  StyledImg,
} from './Picture.styles';

const Picture = ({
  src, srcsets, alt, title, className,
}) => {
  if (!src) return null;
  return (
    <StyledPicture className={className}>
      {srcsets.map((source) => (
        <source
          srcSet={source.srcset}
          media={source.media}
          key={source.srcset}
        />
      ))}
      <StyledImg src={src} alt={alt} title={title} className={className} />
    </StyledPicture>
  );
};

const picturePropTypes = {
  /**
   * src attribute for the img tag
   */
  src: PropTypes.string.isRequired,
  /**
   * srcsets attribute for the picture tag
   */
  srcsets: PropTypes.arrayOf(PropTypes.shape({
    srcset: PropTypes.string,
    media: PropTypes.string,
  })),
  /**
   * alt text
   */
  alt: PropTypes.string,
  /**
   * title text
   */
  title: PropTypes.string,
  /**
   * Classes to be applied to the Picture component
   */
  className: PropTypes.string,
};

Picture.propTypes = picturePropTypes;

Picture.defaultProps = {
  srcsets: [],
  alt: '',
  title: '',
  className: '',
};

export { Picture, picturePropTypes };
