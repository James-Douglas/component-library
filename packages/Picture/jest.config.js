const baseConfig = require('../../jest.config.base');

module.exports = {
  ...baseConfig,
  roots: [
    '<rootDir>/packages/Picture',
  ],
  collectCoverageFrom: [
    '*.{js,jsx,ts,tsx}',
  ],
  name: 'Picture',
  displayName: 'Picture',
  rootDir: '../..',
};
