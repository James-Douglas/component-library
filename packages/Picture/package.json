{
  "name": "@comparethemarketau/manor-picture",
  "version": "0.0.9",
  "description": "Manor Picture component",
  "main": "lib/index.cjs.js",
  "author": "Compare The Market",
  "license": "Proprietary",
  "scripts": {
    "build": "rollup -c '../../rollup.config.js'",
    "tdd": "jest --coverage --watchAll",
    "test": "cross-env JEST_SILENT_REPORTER_DOTS=true jest --reporters=jest-silent-reporter --coverage --watchAll",
    "test:ci": "cross-env CI=true jest",
    "prepare": "cp package.json lib/ && cp CHANGELOG.md lib/"
  },
  "publishConfig": {
    "registry": "https://nexus.comparethemarket.com.au/repository/platform-packages/",
    "access": "public",
    "tag": "latest"
  },
  "dependencies": {
    "prop-types": "^15.7.2"
  },
  "peerDependencies": {
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "styled-components": "^5.0.0"
  }
}
