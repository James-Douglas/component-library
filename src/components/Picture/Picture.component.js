import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import placeholder from 'images/placeholder.png';

const StyledPicture = styled.picture`
  height: 100%;
`;

const StyledImg = styled.img`
  height: 100%;
  max-width: 100%;
  object-fit: contain;
  object-position: left;
`;

const Picture = ({
  src, srcsets, alt, title, className,
}) => (
  <StyledPicture className={className}>
    {srcsets.map((source) => (
      <source
        srcSet={source.srcset}
        media={source.media}
        key={source.srcset}
      />
    ))}
    <StyledImg src={src} alt={alt} title={title} className={className} />
  </StyledPicture>
);

export const picturePropTypes = {
  /**
   * src attribute for the img tag
   */
  src: PropTypes.string,
  /**
   * srcsets attribute for the picture tag
   */
  srcsets: PropTypes.arrayOf(PropTypes.shape({
    srcset: PropTypes.string,
    media: PropTypes.string,
  })),
  /**
   * alt text
   */
  alt: PropTypes.string,
  /**
   * title text
   */
  title: PropTypes.string,
  /**
   * Classes to be applied to the Picture component
   */
  className: PropTypes.string,
};

Picture.propTypes = picturePropTypes;

Picture.defaultProps = {
  src: placeholder,
  srcsets: [],
  alt: 'a placeholder image',
  title: 'a placeholder title',
  className: '',
};

export default Picture;
