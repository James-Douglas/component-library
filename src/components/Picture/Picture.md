# Icon component
Basic responsive picture component
## Usage

~~~js
<Picture
  src
  srcsets
  alt
  title
/>
~~~

## Props

1. `src` *(optional) string*
    * The src of the image.
2. `srcsets` *(optional) string array*
	* Array of image sources along with widths as per https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#Resolution_switching_Different_sizes
3. `alt` *(optional) string*
	* Image alt text
4. `title` *(optional) string*
    * Image title text
