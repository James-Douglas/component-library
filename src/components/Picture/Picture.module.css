picture {
  height: 100%;
}

img {
  height: 100%;
  object-fit: contain;
  object-position: left;
}
