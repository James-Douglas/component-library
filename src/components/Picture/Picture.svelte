<svelte:options tag='ctm-picture' />
<picture>
  {#each srcsets as data }
  <source {...data} />
  {/each}
  <img {src} {alt} {title} />
</picture>

<style>

  picture {
    height: 100%;
  }

  img {
    height: 100%;
    object-fit: contain;
    object-position: left;
  }
</style>

<script>
  export let style = '';
  export let src = 'https://via.placeholder.com/249x44';
  export let srcsets = [];
  export let alt = 'a placeholder image';
  export let title = 'a placeholder title';
</script>
