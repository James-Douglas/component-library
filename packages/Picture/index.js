import { Picture, picturePropTypes } from './Picture.component';

export { Picture, picturePropTypes };
