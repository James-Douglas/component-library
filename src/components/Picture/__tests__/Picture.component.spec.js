import React from 'react';
import { render } from '@testing-library/react';
import Picture from '../Picture.component';

describe('Picture', () => {
  it('renders correctly without props', () => {
    const { container } = render(<Picture />);
    expect(container).toMatchSnapshot();
    const img = container.querySelector('img');
    expect(img).toBeDefined();
    expect(img.getAttribute('alt')).toEqual('a placeholder image');
    expect(img.getAttribute('src')).toEqual('placeholder.png');
    expect(img.getAttribute('title')).toEqual('a placeholder title');
  });

  it('uses title and alt props', () => {
    const { container } = render(<Picture alt="test" title="Test" />);
    const img = container.querySelector('img');
    expect(img).toBeDefined();
    expect(img.getAttribute('alt')).toEqual('test');
    expect(img.getAttribute('title')).toEqual('Test');
  });
});
