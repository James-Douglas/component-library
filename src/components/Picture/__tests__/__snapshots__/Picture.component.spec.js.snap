// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture renders correctly without props 1`] = `
<div>
  <picture>
    <img
      alt="a placeholder image"
      src="placeholder.png"
      title="a placeholder title"
    />
  </picture>
</div>
`;
